### ISSUE TITLE
hmget returns false instead of an array (phpredis 3.0.0)

### ISSUE THREAD

**Issue by blat** [phpredis/phpredis#946]
With PHP 7.0.10 and PHP Redis 3.0.0 :

```
$redis = new Redis();
$redis->connect('127.0.0.1', 6379);

$key = 'foobar';
$fields = ['foo', 'bar'];
foreach ($fields as &$f) {}
$data = $redis->hmget($key, $fields);
var_dump($data); // bool(false)

$redisCluster = new RedisCluster(null, [
   '127.0.0.1:7001',
   '127.0.0.1:7002',
   '127.0.0.1:7003',
]);

$key = 'foobar';
$fields = ['foo', 'bar'];
foreach ($fields as &$f) {}
$data = $redisCluster->hmget($key, $fields);
var_dump($data); // bool(false)
```

Just remove the `&`, it works:

```
$key = 'foobar';
$fields = ['foo', 'bar'];
foreach ($fields as $f) {}
$data = $redis->hmget($key, $fields);
var_dump($data); /* array(2) {
  ["foo"]=>
  string(4) "blue"
  ["bar"]=>
  string(2) "42"
} */

$key = 'foobar';
$fields = ['foo', 'bar'];
foreach ($fields as $f) {}
$data = $redisCluster->hmget($key, $fields);
var_dump($data); /* array(2) {
  ["foo"]=>
  string(4) "blue"
  ["bar"]=>
  string(2) "42"
} */
```

Not reproducible with PHP 5.6.24 and PHP Redis 2.2.8.

**Comment by yatsukhnenko** [comment]
php7 differently works with references. Possible fix is
```
diff --git a/redis_commands.c b/redis_commands.c
index 61c6556..436a220 100644
--- a/redis_commands.c
+++ b/redis_commands.c
@@ -1493,6 +1493,7 @@ int redis_hmget_cmd(INTERNAL_FUNCTION_PARAMETERS, RedisSock *redis_sock,
     // Iterate over our member array
     ZEND_HASH_FOREACH_VAL(ht_arr, z_mem) {
         // We can only handle string or long values here
+        if (Z_ISREF_P(z_mem)) ZVAL_DEREF(z_mem);
         if ((Z_TYPE_P(z_mem) == IS_STRING && Z_STRLEN_P(z_mem) > 0)
             || Z_TYPE_P(z_mem) == IS_LONG)
         {
```

**Comment by Skunnyk** [comment]
Works with your fix on php7 with 3.0.0 release and git branch php7 :)

**Comment by eranhazout** [comment]
Did you fix this bug with php 7.0?
I still receive boolean instead of array, see usage: 
`        $multi = $redis->multi(Redis::PIPELINE);
        foreach($ids as $id){
            $cache_key = 'fields.' . $id;
            $multi->hMGet($cache_key,$fields);
        }
        $data = $multi->exec();`

We are currently receiving many errors because of it.
Any suggestions?
